lldb/source/Plugins/Process/elf-core/ProcessElfGpuCore.cpp

```diff
@@ -0,0 +1,165 @@
+//===-- ProcessElfGpuCore.cpp
+//------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstdlib>
+
+#include <memory>
+
+#include "ProcessElfGpuCore.h"
+
+#include "Plugins/DynamicLoader/POSIX-DYLD/DynamicLoaderPOSIXDYLD.h"
+#include "lldb/Core/Debugger.h"
+#include "lldb/Core/Module.h"
+#include "lldb/Core/PluginManager.h"
+#include "lldb/Target/DynamicLoader.h"
+#include "lldb/Utility/LLDBLog.h"
+#include "lldb/Utility/Log.h"
+#include "llvm/Support/Threading.h"
+
+using namespace lldb_private;
+using namespace lldb;
+
+lldb::ProcessSP ProcessElfGpuCore::CreateInstance(lldb::TargetSP target_sp,
+                                                  lldb::ListenerSP listener_sp,
+                                                  const FileSpec *crash_file,
+                                                  bool can_connect) {
+  return ProcessElfCore::CreateInstance(target_sp, listener_sp, crash_file,
+                                        can_connect);
```

Comment:
All of the work in `ProcessElfGpuCore::FindGpuPlugin` should happen here. If everythong goes well, then we can return a valid `ProcessSP`, and if not, return an empty ProcessSP. The PRocessELFCore should call the new CreateInstance function:
```
typedef lldb::ProcessSP (*ELFEmbeddedCoreCreateInstance)(ProcessELFCore *native_core, lldb::ListenerSP listener_sp);
```
And then all of the work done in ProcessElfGpuCore::FindGpuPlugin should be done here.